### ISSUE TITLE
sphinx in environment.yml pip section make poetry solver run forever (seems solved with poetry 1.2)

### ISSUE THREAD

**Issue by lesteve** [conda/conda-lock#309]
See in a scikit-learn context see https://github.com/scikit-learn/scikit-learn/pull/25342#issuecomment-1377262962. I tried with a range of conda-lock versions: 1.2.1, 1.3.0 and from `main` and see the same behaviour.

```
conda-lock lock --log-level DEBUG -f environment.yml -p linux-64
```

```yml
channels:
  - conda-forge
dependencies:
  - python
  - pip
  - pip:
    - sphinx
```

The last logged output is:
```
DEBUG:conda_lock._vendor.poetry.repositories.pypi_repository:<debug>PyPI:</debug> 4 packages found for sphinxcontrib-applehelp *
```

I can reproduce the hang with only `sphinxcontrib-applehelp` in the pip section.

I can reproduce the same behaviour with this Python code mirroring what conda-lock does:

```py
from conda_lock.src_parser import VersionedDependency
from conda_lock.pypi_solver import solve_pypi

pip_specs = {
    "sphinxcontrib-applehelp": VersionedDependency(
        name="sphinxcontrib-applehelp", manager="pip", version="*"
    )
}
solve_pypi(
    pip_specs,
    use_latest=[],
    pip_locked={},
    conda_locked={},
    python_version="3.10",
    platform="linux-64",
    verbose=True
)
```

which just keeps printing:
```
<debug>   1:</debug> selecting sphinxcontrib.applehelp (1.0.3)
```

Maybe this is because `sphinxcontrib-applehelp` (with a dash) on PyPI redirects to `sphinxcontrib.applehelp` (with a dot) and somehow this aliases mechanism is not taken into account by conda-lock or poetry.

I am not very familiar with poetry, but it seems like if I add `sphinxcontrib-applehelp` to `pyproject.toml`
- I get a similar hang with poetry 1.1.15
- it works with poetry 1.2.0 

So maybe the fix would be to require a more recent version of poetry and/or update the vendored poetry code. Not sure how feasible this is in conda-lock.

Side-comment: not sure what has changed, it used to work fine maybe 1-2 months ago, it does not seem related to new sphinxcontrib-applehelp releases (I tried pinning to older releases), the only thing I can come up with is that the PyPI redirect sphinxcontrib-applehelp -> sphinxcontrib.applehelp has happened recently?

**Comment by mariusvniekerk** [comment]
Sphinx is very much available on conda-forge though...

**Comment by lesteve** [comment]
Fair point, but in one of the scikit-learn builds we run our CI against the latest versions of what is available with pip ...

**Comment by lesteve** [comment]
It seems like this has been fixed somehow in conda-lock 2.0.0, closing
